packages/jest-snapshot/src/getMatchers.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const common = require('./common');
+const fs = require('fs');
+const path = require('path');
+const serializer = require('./serializer');
+const Snapshot = require('./Snapshot');
+
+const testsByPath =  {};
+const getNextIndexFor = (test, filePath) => {
+  const tests = testsByPath[filePath] || (testsByPath[filePath] = {});
+  if (test in tests) {
```

Comment:
this shouldn't use `in` because it looks up the prototype chain. It should either use Object.create above or check whether `tests[test] != null`.